src/Avalonia.Styling/Controls/ResourceNodeExtensions.cs

```diff
@@ -145,15 +178,44 @@ protected override void Deinitialize()
 
             protected override void Subscribed(IObserver<object?> observer, bool first)
             {
-                if (_target.Owner is object)
+                if (_target.Owner is IResourceHost owner)
                 {
-                    observer.OnNext(Convert(_target.Owner?.FindResource(_key)));
+                    var (found, hostAttachedToStylingTree) = owner.TryFindResourceWithParentCheck(_key, out var value);
+                    if (!found)
+                    {
+                        value = AvaloniaProperty.UnsetValue;
+
+                        if (hostAttachedToStylingTree == true)
+                        {
+                            Logger.TryGet(LogEventLevel.Warning, LogArea.Binding)?.Log(this, "Warning: Dynamic resource '{Key}' was not found in the {Owner}.", _key, owner.GetType().Name);
+                        }
+                    }
+
+                    observer.OnNext(Convert(value));
                 }
             }
 
             private void PublishNext()
             {
-                PublishNext(Convert(_target.Owner?.FindResource(_key)));
```

Comment:
@grokys I believe that line was a reason of "cannot convert null to color" error messages in the logging output.
When user is changing theme, target.Owner is null for some resource providers and null value published to the observer. I added null check with publishing UnsetValue on null. Should I create separated PR just for that fix?